Jetpack Social: Update the strings to reflect the new publicize branding

enejb opened the pull request:
Updates the Apps to have make Jetpack Social more predominant in the apps.
See pdrWKz-vt-p2

|Before | After |
|---|---|
|![Screenshot_20220901_151100](https://user-images.githubusercontent.com/17463767/187887910-3f704bd6-57cf-4e8a-91cd-c219f67baef9.png)|![Screenshot_20220901_152255](https://user-images.githubusercontent.com/17463767/187887917-bc89e361-00c4-4614-a7be-eb3c537970cd.png)|
|![Screenshot_20220901-153201_WordPress Pre-Alpha](https://user-images.githubusercontent.com/17463767/187888275-542e9900-f1ae-49a4-a2e8-9f59de9d12b8.jpg)|![Screenshot_20220901_152311](https://user-images.githubusercontent.com/17463767/187887922-56a2114b-56b0-4c9c-ae3e-6f747f1881bc.png)


### To test:

**Test 1**
Go to Site > Settings > Sharing 
Notice the new label. 
Add a social account. Notice it says share post to %.

**Test 2**
Go create a new post. 
Select Post settings. 
Notice it says Jetpack Social as a heading for the social connections.

## Regression Notes
1. Potential unintended areas of impact
No regressions this is just a text change.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Looked at the UI in the PR. 

3. What automated tests I added (or what prevented me from doing so)
none.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

enejb committed:
Jetpack Social: Update the strings to reflect the new publicize branding

AjeshRPai reviewed (approved):
@enejb I have tested the changes. All looks fine to me 👍🏼 

I am not sure whether the changes to `fastlane/resources/values/strings.xml` is needed or not. 🤔 . Maybe I am missing something. I will let @zwarm confirm it. 


>Go to Site > Settings > Sharing
Notice the new label.
Add a social account. Notice it says share post to %.

🟢  Works as expected

>Test 2
Go create a new post.
Select Post settings.
Notice it says Jetpack Social as a heading for the social connections.

 I couldn't test this, am not sure whether these are the right steps or am missing something. 🤔 

As such I don't see anything blocking. I am approving but not merging so that @zwarm can have a look.

zwarm commented:
@enejb 
Same as @AjeshRPai , I could not follow the steps "Test 2" - I don't see a sharing option.

And @AjeshRPai is also correct, we don't change the fastlane files, just the `strings.xml`. So please revert that commit. 🙏 .

❓ Just to be double check, you want the text to read `Jetpack Social Connections` in both the WordPress app and the Jetpack app, correct? If this is not the case, then you need to add the value to the Jetpack [strings.xml](WordPress/src/jetpack/res/values/strings.xml) instead of the WordPress [strings.xml](WordPress/src/main/res/values/strings.xml) file. Adding it to the WordPress file will change it for both apps.

zwarm reviewed (changes requested):
Please revert the fastlane commit. It is not necc. to directly change those file. Thanks.

enejb committed:
Revert fastlane chanes

enejb committed:
Minor fixed. Remove the possessive pronoun

enejb commented:
@zwarm and @AjeshRPai I reverted the changes to fast lane and got more feedback Kristina Plauché. The chance should be the same on Jetpack and WP apps. I also updated wording from `Share your post %` to just `Share post %`. This way it lines up with iOS as well. 

I think this is ready to merge now. But please take another look in case I missed something

AjeshRPai reviewed (approved):
Everything LGTM 👍🏼 . @zwarm Since you requested changes, GitHub is not showing me the option to merge the PR. Once you have a look at the changes, please feel free to merge.

zwarm reviewed (approved):
Nice work @enejb . Thanks for reverting the fastlane file. :shipit:
